src/pages/docs/creators/javascript/tutorials/using-reputation.md

```diff
@@ -78,6 +78,32 @@ reputation.setProposalWeights({
 
 In the example provided, the uptime metric is given a higher weight (0.7) than the success rate (0.3). This means that the system will favor providers with better uptime. These weights are flexible and can be adjusted according to the specific needs of your project. By experimenting with these weights, you can find the optimal balance and select providers that best meet your project's requirements.
 
+### Maximizing Performance with the Agreement Selector
+
+The agreement selector is an additional filter that helps you choose the best providers for your tasks. It works in tandem with our proposal filter, to achieve optimal performance.
+
+- **Targeted Filtering:** Our `ProposalFilter` acts as the initial screening tool. It closely examines all available providers, evaluating them against the weights you've defined. Providers that don't meet your weights are eliminated, leaving behind a list of qualified candidates.
+
+- **Precision Selection:** The `agreementSelector` takes over from there. It focuses on the list of providers vetted by our `proposalFilter`. By closely analyzing their performance data, which includes historical single- and multithread CPU benchmark scores, the agreement selector identifies the absolute best fit for your task.
+
+Imagine the proposal filter as a sieve. It sorts through all the available providers, leaving only those that meet your requirements. Then, the agreement selector steps in. It acts like a magnifying glass, examining the remaining good providers closely to find the absolute best fit for your specific task.
+
+The agreement selector doesn't just pick the provider with the highest overall score. Instead, it considers a smaller group of top performers, chosen based on a setting called `topPoolSize`.
+
+#### Why topPoolSize Matters
+
+This setting influences the number of high-ranking providers included in the final selection pool. Here's a breakdown of the impact of different topPoolSize values:
+
+- **topPoolSize set to 1:** Choosing only the single highest-ranked provider might guarantee exceptional results, but it limits your options. For instance, if you have multiple tasks running concurrently, you might end up using the same provider repeatedly, overlooking other well-suited providers.
+
+- **topPoolSize set to a higher number (like 5 or 8):** With a larger pool, you have more flexibility. However, an excessively high number can weaken the selection process, especially when dealing with a vast number of providers. In such cases, the impact our `proposalFilter` weights might diminish, and the selection might become more random. A larger pool size is only advantageous when there's a significantly larger pool of qualified providers compared to the final selection pool size.
```

Comment:
"especially when dealing with a vast number of providers" - that's the actually the inverse of what I'm observing. `topPoolSize / availableCandidate` the lower this number, the better.

Also this highly depends on score distribution. Consider the following cases with given scores (higher is better):
Case 1: 10, 5, 2, 1, 1, 1, 1, 0, 0, 0
Case 2: 10, 9, 9, 8, 7, 7, 6, 4, 4, 4
What will be your candidates in the above two cases for topPool size of 2, 4, and 6